### ISSUE TITLE
Rule 1200 message not helpful when developer setup includes async result

### ISSUE THREAD

**Issue by rjmurillo** [Litee/moq.analyzers#4]
When using the result of an async operation (returning generic `Task<>`), rule Moq1200 fires indicating setup should only be used for overridable types. This is technically true because `Task<>` is abstract, however, the message is not helpful / actionable by the developer.

Example

```csharp
public class AsyncClient
{
    public virtual Task<string> GenericAsyncWithConcreteReturn() => Task.FromResult(string.Empty);
}

var mock = new Mock<AsyncClient>();

// Error 1200
mock.Setup(c => c.GenericAsyncWithConcreteReturn().Result);

// Expected 
mock.Setup(c => c.GenericAsyncWithConcreteReturn())
                .ReturnsAsync(string.Empty);
```

Propose adding a new rule, Moq1201, that is triggered when the setup method includes a Task that calls in to `.Result`. The error message reflects the guidance from Moq, which is to use the `.ReturnsAsync` method for setup.

**Comment by Litee** [comment]
Hi @rjmurillo. I am not using .NET and Moq nowdays. Do you want to take over this package?

**Comment by rjmurillo** [comment]
@Litee Sure, thanks!

**Comment by Litee** [comment]
Let me know your user name in NuGet Gallery, I will add you as an owner.

**Comment by rjmurillo** [comment]
It's the same: [`rjmurillo`](https://www.nuget.org/profiles/rjmurillo)

**Comment by Litee** [comment]
I've added you as a collaborator for two repositories and NuGet packages. Let me know if you have any issues with publishing those updates.

**Comment by rjmurillo** [comment]
@Litee Would you kindly resend the GitHub collaborate request?

**Comment by Litee** [comment]
Re-sent.
